### PR TITLE
fix: Handle mov edge case (Quicktime mov)

### PR DESCRIPTION
**Pull request by tmathern**
## Changes in this pull request
With release 0.75.18 (I skipped 0.75.17), tests in c2pa-python for mov files started failing:
```
ERROR: test_sign_mov_video_file_single (__main__.TestBuilderWithSigner)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/c2pa-python/c2pa-python/./tests/test_unit_tests.py", line 2774, in test_sign_mov_video_file_single
    builder.sign(self.signer, "mov", file, output)
  File "/home/runner/work/c2pa-python/c2pa-python/src/c2pa/c2pa.py", line 3105, in sign
    manifest_bytes = self._sign_internal(
  File "/home/runner/work/c2pa-python/c2pa-python/src/c2pa/c2pa.py", line 3043, in _sign_internal
    error = _parse_operation_result_for_error(_lib.c2pa_error())
  File "/home/runner/work/c2pa-python/c2pa-python/src/c2pa/c2pa.py", line 792, in _parse_operation_result_for_error
    _raise_typed_c2pa_error(error_str)
  File "/home/runner/work/c2pa-python/c2pa-python/src/c2pa/c2pa.py", line 745, in _raise_typed_c2pa_error
    raise C2paError.Other(error_str)
c2pa.c2pa._C2paOther: Other: asset could not be parsed: Box size extends beyond asset bounds

```
(Example: https://github.com/contentauth/c2pa-python/actions/runs/21790410303/job/62874147265?pr=226)

This is the test fixture: https://github.com/contentauth/c2pa-python/blob/vNext/tests/fixtures/C-recorded-as-mov.mov
The fixture was created by making a screen recording from one of the images (so it's an Apple MOV). 

I believe we missed an edge case here, and this fixes that (verified to work with the c2pa-python tests).
Dug into it with a lot of tooling to find the difference in files and handling of spec.

## Checklist
- [x] This PR represents a single feature, fix, or change.
- [x] All applicable changes have been documented.
- [x] Any `TO DO` items (or similar) have been entered as GitHub issues and the link to that issue has been included in a comment.

### CODE REVIEW

File: sdk/tests/test_reader.rs [hunk 1]
```diff
@@ -58,17 +58,8 @@ fn test_reader_xca_jpg() -> Result<()> {
...
-        reader
-            .validation_results()
-            .unwrap()
-            .active_manifest()
-            .unwrap()
-            .failure[0]
-            .code(),
-        validation_status::ASSERTION_DATAHASH_MISMATCH
-    );
+    // Validation failures can appear in any order (e.g. signingCredential.untrusted vs assertion.dataHash.mismatch).
```
Comment (tmathern) [inline sdk/tests/test_reader.rs]: Unflakes that test
